cesium/internal/unary/writer.go

```diff
@@ -142,12 +154,12 @@ func (db *DB) OpenWriter(ctx context.Context, cfgs ...WriterConfig) (w *Writer,
 		}, err
 	})
 	if err != nil {
-		return nil, transfer, err
+		return nil, transfer, w.wrapError(err)
 	}
 
 	w.control = g
-	db.mu.Add(1)
-	return w, transfer, err
+	db.mu.add(1)
```

Comment:
Hard to figure out why you're adding a value hear. Might be better to name this function something like `addOpenWriterIterator`

Reply:
Since the mutex only controls the entity count, I think it's better to just rename the variable from mu to entityCount, so we do `db.entityCount.Add`, reminiscent of something like `wg.Add(1)` or `sem.Add(1)`.